### PR TITLE
#428 Add a Windows Events profile

### PR DESCRIPTION
**Pull request by mr-splunk**
Adds the long-awaited Windows Events profile. Kept it short and simple to start:

image

### CODE REVIEW

File: events/system/windows/registry_key.json [hunk 1]
```diff
@@ -5,6 +5,9 @@
   "name": "registry_key_activity",
   "uid": 8,
   "attributes": {
+    "$include": [
```
Comment (rroupski) [inline events/system/windows/registry_key.json:8]: The `registry_key_activity` extends the `system` class, so you don't need to include the `winlog` profile.

**Review by rroupski — CHANGES_REQUESTED** [review]

File: dictionary.json [hunk 2]
```diff
@@ -1331,6 +1331,11 @@
       "description": "Error Message",
       "type": "string_t"
     },
+    "event_code": {
+      "description": "The Event ID or Code that the product uses to describe the event. Ex: EventCode=4624 (multiline) or EventID=4624 (xml).",
+      "requirement": "required",
+      "type": "string_t"
+    },
```
Comment (pagbabian-splunk) [inline dictionary.json:1338]: I would avoid putting requirements in the dictionary, as they will be defined by how they are used in each class or profile.

**Comment by pagbabian-splunk** [comment]
I thought we would have platform naming, e.g. Windows, rather than WinLog which is the caption in the code (vs. Windows Events which is the popup caption in the screen shot).  Also, I would avoid camel case as we don't use it anywhere.  Can we make the profile name `Windows`?

**Comment by awhite456** [comment]
> I thought we would have platform naming, e.g. Windows, rather than WinLog which is the caption in the code (vs. Windows Events which is the popup caption in the screen shot). Also, I would avoid camel case as we don't use it anywhere. Can we make the profile name `Windows`?

This profile really only adds support for Windows Event Log messages, it doesn't really provide any general Windows functionality. I think we discussed this on the call before you were able to join

File: profiles/winlog.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "log_name": {
+      "description": "The Windows logging subsystem. Ex: Channel=Security (xml) or LogName=Security (multiline)",
+      "requirement": "recommended"
+    },
+    "log_version": {
+      "description": "The version of the event log template. Ex: Version=1 (xml). Note: Multiline formatted windows events do not record this attribute. For more information, refer to 'Event Versions' in <a target='_blank' href='https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4624'>this page</a>.",
+      "requirement": "optional"
+    },
+    "provider": {
+      "description": "The service that logged the Windows event. Ex: 'Microsoft-Windows-Security-Auditing' (xml) or Microsoft Windows security auditing. (multiline).",
```
Comment (awhite456) [inline profiles/winlog.json:24]: Nit - Either both examples should have quotes or neither

File: dictionary.json (hunk 2 above)
Comment (awhite456) [reply dictionary.json:1337]: +1, shouldn't have requirements here

**Review by awhite456 — COMMENTED** [review]
Looks good once other feedback is addressed

File: events/system/windows/registry_key.json (hunk 1 above)
Comment (mr-splunk) [reply events/system/windows/registry_key.json:8]: Removed the winlog include

File: profiles/winlog.json (hunk 3 above)
Comment (mr-splunk) [reply profiles/winlog.json]: Resolved via fa674a279f312dfb30dbe78101813f58ccd3278f

**Review by awhite456 — APPROVED** [review]
lgtm

File: dictionary.json [hunk 4]
```diff
@@ -1853,6 +1857,14 @@
       "description": "The detailed geographical location usually associated with an IP address.",
       "type": "location"
     },
+    "log_name": {
+      "description": "The logging subsystem. Ex: Channel=Security (xml) or LogName=Security (multiline)",
+      "type": "string_t"
+    },
+    "log_version": {
+      "description": "The version of the event log template. Ex: Version=1 (xml). Note: Multiline formatted windows events do not record this attribute. For more information, refer to 'Event Versions' in <a target='_blank' href='https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4624'>this page</a>.",
+      "type": "integer_t"
```
Comment (pagbabian-splunk) [inline dictionary.json:1866]: Can we move the very specific Windows description to the profile specific use and make the dictionary description more general purpose?

File: profiles/winlog.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+  "description": "The attributes that identify Windows Event Log entries.",
+  "meta": "profile",
+  "caption": "WinLog",
```
[the remark this answers is not in the document]
Comment (mr-splunk) [reply profiles/winlog.json]: Changed via 21ece1036c242038519d28dd0e03bf923a305eca

File: dictionary.json (hunk 4 above)
Comment (mr-splunk) [reply dictionary.json:1866]: Take a look at changes on this commit: 38a312db06277aab5706a9da2f8d4372b487c005

**Review by pagbabian-splunk — APPROVED** [review]

File: dictionary.json [hunk 6]
```diff
@@ -1853,6 +1857,14 @@
       "description": "The detailed geographical location usually associated with an IP address.",
       "type": "location"
     },
+    "log_name": {
+      "description": "The logging subsystem the product uses for the event. See specific usage.",
+      "type": "string_t"
+    },
+    "log_version": {
+      "description": "The version of the event log template which the product uses for the event. See specific usage.",
+      "type": "integer_t"
```
Comment (rroupski) [inline dictionary.json:1866]: to make the version useful for other products (general purpose), the type should be string, not an integer.

**Review by rroupski — CHANGES_REQUESTED** [review]

File: dictionary.json (hunk 6 above)
Comment (mr-splunk) [reply dictionary.json]: Changed to `string_t` via commit 70dc6e26d73f25cf2d74846a701c59ab5afef3af

**Review by rroupski — APPROVED** [review]

File: dictionary.json (hunk 6 above)
Comment (pagbabian-splunk) [reply dictionary.json]: Good catch.

**Review by pagbabian-splunk — APPROVED** [review]

**Review by jp-harvey — APPROVED** [review]
LGTM

**Pull request merged**